Fix documentation for the required argument

kumabotz opened the pull request:
## Description

Fix documentation for the required `focusNode` argument

## Related Issues

#41475

## Tests

No test added as this is a documentation fix,

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

kumabotz committed:
Fix documentation for the required argument

goderbauer commented on packages/flutter/lib/src/widgets/raw_keyboard_listener.dart:
```diff
@@ -46,6 +46,8 @@ class RawKeyboardListener extends StatefulWidget {
        assert(child != null),
        super(key: key);
 
+  /// Controls whether this widget has keyboard focus.
```
The macro below contains the documentation. In the docs it looks like this: https://master-api.flutter.dev/flutter/widgets/RawKeyboardListener/focusNode.html

This change doesn't seem to add more meaningful documentation than what's already there?

kumabotz replied:
The changes is to increase clarity when reading the documentation from https://api.flutter.dev/flutter/widgets/RawKeyboardListener-class.html according to the issue reported from #41475

goderbauer replied:
Even with this change the documentation for the property will still say that the focus node is optional because of the macro even though its marked as required in the constructor. 

/cc @gspencergoog

kumabotz replied:
@goderbauer , I've removed the macro to make the documentation less confusing.

Piinks replied:
We make use of macros for documenting classes in many places throughout the code base. @kumabotz, do you find them confusing in general or just in this specific case?

goderbauer replied:
My understanding is that the macro is wrong. The macro claims this property is optional and talks about what it means to set or not set it, but the constructor has this property actually marked as required. Not setting it is not an option.

kumabotz committed:
Remove the confusing macro documentation that states the property is
optional

A user can still learn more about the FocusNode by clicking on
the class name
